### PR TITLE
raftstore-v2: fix paused checkpoint and other issues

### LINKED ISSUE #15059: [Dynamic Region] raft log lag more and more after inject one tikv network partition for 50m and recover
### What version of TiKV are you using?
./tikv-server -V
 TiKV 
Release Version:   7.3.0-alpha
Edition:           Community
Git Commit Hash:   425f6f262f577425612cf1663ea6cceada72c724
Git Commit Branch: heads/refs/tags/v7.3.0-alpha
UTC Build Time:    2023-06-30 11:15:31
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release
2023-07-03T09:59:28.474+0800

### What operating system and CPU are you using?
8c/32g

### Steps to reproduce
1、run tpcc
2、inject one tikv network partition for 50m and recover

### What did you expect?
raft log lag can catch up within some time

### What did happened?
raft log lag more and more after inject one tikv network partition for 50m and recover
9f12f8e1-48e0-431b-80b2-b89c02a19250

### LINKED ISSUE #15181: [Dynamic Region] tikv oom when simulate fault testing due to the block cache configuration is too large
### What version of TiKV are you using?
./tikv-server -V
 TiKV 
Release Version:   7.3.0-alpha
Edition:           Community
Git Commit Hash:   2f2900a6ff1fa0b7c881718668c1714ecfdcbc2f
Git Commit Branch: heads/refs/tags/v7.3.0-alpha
UTC Build Time:    2023-07-20 11:12:29
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release

### What operating system and CPU are you using?
8c/32g

### Steps to reproduce
tikv oom when do some fault testing such as：
     simulate network partition between one tikv and other all pods
     simulate network partition between pd leader and other all pods
     simulate one of tikv failure for 10mins

### What did you expect?
no oom

### What did happened?
tikv oom when do some fault testing 
image

### LINKED ISSUE #15209: [Dynamic Regions] tikv panic for "[FATAL] [lib.rs:510] ["source peer is missing when getting checkpoint for merge [peer_id=6259] [region_id=6258]"
### What version of TiKV are you using?
[2023/07/24 09:11:35.904 +00:00] [INFO] [lib.rs:93] ["Git Commit Hash:   aa3a0b014296b5ee42b88fd67f7ac564a01b60bb"]

### Steps to reproduce
jepsen : monotonic-mixed-follower-syserr test 
 "Establishes a collection of integers identified by keys. Monotonically
  increments individual keys via read-write transactions, and reads keys in
  small groups. We verify that the order of transactions implied by each key
  are mutually consistent; e.g. no transaction can observe key x increase, but
  key y decrease."
### What did happened?
```
[2023/07/24 09:15:55.408 +00:00] [FATAL] [lib.rs:510] ["source peer is missing when getting checkpoint for merge [peer_id=6259] [region_id=6258]"] [backtrace="   0: tikv_util::set_panic_hook::{{closure}}\n             at /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/tikv_util/src/lib.rs:509:18\n   1: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2032:9\n      std::panicking::rust_panic_with_hook\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:692:13\n   2: std::panicking::begin_panic_handler::{{closure}}\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:579:13\n   3: std::sys_common::backtrace::__rust_end_short_backtrace\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys_common/backtrace.rs:137:18\n   4: rust_begin_unwind\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:575:5\n   5: core::panicking::panic_fmt\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:65:14\n   6: raftstore_v2::operation::command::admin::merge::commit::<impl raftstore_v2::raft::apply::Apply<EK,R>>::apply_commit_merge::{{closure}}\n             at /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/raftstore-v2/src/operation/command/admin/merge/commit.rs:390:25\n      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/future/mod.rs:91:19\n      raftstore_v2::operation::command::<impl raftstore_v2::raft::apply::Apply<EK,R>>::apply_entry::{{closure}}\n             at /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/raftstore-v2/src/operation/command/mod.rs:697:91\n   7: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/future/mod.rs:91:19\n      raftstore_v2::operation::command::<impl raftstore_v2::raft::apply::Apply<EK,R>>::apply_committed_entries::{{closure}}\n             at /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/raftstore-v2/src/operation/command/mod.rs:576:61\n      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/future/mod.rs:91:19\n      raftstore_v2::fsm::apply::ApplyFsm<EK,R>::handle_all_tasks::{{closure}}\n             at /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/raftstore-v2/src/fsm/apply.rs:148:94\n   8: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/future/mod.rs:91:19\n      raftstore_v2::operation::command::<impl raftstore_v2::raft::peer::Peer<EK,ER>>::schedule_apply_fsm::{{closure}}\n             at /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/raftstore-v2/src/operation/command/mod.rs:162:61\n      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/future/mod.rs:91:19\n      <tracker::tls::TrackedFuture<F> as core::future::future::Future>::poll::{{closure}}\n             at /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/tracker/src/tls.rs:64:23\n      std::thread::local::LocalKey<T>::try_with\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/local.rs:446:16\n      std::thread::local::LocalKey<T>::with\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/local.rs:422:9\n      <tracker::tls::TrackedFuture<F> as core::future::future::Future>::poll\n             at /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/tracker/src/tls.rs:62:9\n      tikv_util::yatp_pool::future_pool::PoolInner::spawn::{{closure}}\n             at /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/tikv_util/src/yatp_pool/future_pool.rs:173:27\n      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/future/mod.rs:91:19\n      yatp::task::future::RawTask<F>::poll\n             at /rust/git/checkouts/yatp-e704b73c3ee279b6/5572a78/src/task/future.rs:59:9\n   9: yatp::task::future::TaskCell::poll\n             at /rust/git/checkouts/yatp-e704b73c3ee279b6/5572a78/src/task/future.rs:103:9\n      <yatp::task::future::Runner as yatp::pool::runner::Runner>::handle\n             at /rust/git/checkouts/yatp-e704b73c3ee279b6/5572a78/src/task/future.rs:387:20\n  10: <tikv_util::yatp_pool::YatpPoolRunner<T> as yatp::pool::runner::Runner>::handle\n             at /home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/tikv_util/src/yatp_pool/mod.rs:196:24\n      yatp::pool::worker::WorkerThread<T,R>::run\n             at /rust/git/checkouts/yatp-e704b73c3ee279b6/5572a78/src/pool/worker.rs:48:13\n      yatp::pool::builder::LazyBuilder<T>::build::{{closure}}\n             at /rust/git/checkouts/yatp-e704b73c3ee279b6/5572a78/src/pool/builder.rs:114:25\n      std::sys_common::backtrace::__rust_begin_short_backtrace\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys_common/backtrace.rs:121:18\n  11: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/mod.rs:551:17\n      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panic/unwind_safe.rs:271:9\n      std::panicking::try::do_call\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:483:40\n      std::panicking::try\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:447:19\n      std::panic::catch_unwind\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panic.rs:137:14\n      std::thread::Builder::spawn_unchecked_::{{closure}}\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/mod.rs:550:30\n      core::ops::function::FnOnce::call_once{{vtable.shim}}\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:513:5\n  12: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2000:9\n      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2000:9\n      std::sys::unix::thread::Thread::new::thread_start\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys/unix/thread.rs:108:17\n  13: start_thread\n  14: clone\n"] [location=/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/raftstore-v2/src/operation/command/admin/merge/commit.rs:390] [thread_name=apply-1]
```

### PR DESCRIPTION
**Pull request by tabokie**
### What is changed and how it works?

Issue Number: Close #15059 Fix #15181 Fix #15209

What's Changed:

```commit-message
- Fix checkpoint being paused forever.
- Abort prepare merge if target is merging another.
- Fix block-cache size is set to sum of cf's cache size by default.
- Fix tablet being asked to destroy twice after commit merge.
- Fix life.rs using the wrong fields of `CommitMergeRequest` to rollback.
- Disable RocksDB stats dumping altogether.
- Shutdown high priority pool manually before drop.
- Add catch up logs into the prepare merge state machine to fix some corner case panics.
```

### Related changes

https://github.com/tikv/rocksdb/pull/342
https://github.com/pingcap/docs/pull/14129

### Check List 

Tests
- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -207,6 +207,8 @@ procinfo = { git = "https://github.com/tikv/procinfo-rs", rev = "6599eb9dca74229
 # After the PR to rust-rocksdb is merged, remember to comment this out and run `cargo update -p rocksdb`.
 # [patch.'https://github.com/tikv/rust-rocksdb']
 # rocksdb = { git = "https://github.com/your_github_id/rust-rocksdb", branch = "your_branch" }
+[patch.'https://github.com/tikv/rust-rocksdb']
+rocksdb = { git = "https://github.com/tabokie/rust-rocksdb", branch = "230720-fix-checkpoint-again" }
```
Comment (overvenus) [inline Cargo.toml:211]: Please use tikv/rust-rocksdb, could you send PR to tikv/rust-rocsdb first?

Comment (tabokie) [reply Cargo.toml:211]: https://github.com/tikv/rocksdb/pull/342

File: components/raftstore-v2/src/worker/tablet.rs [hunk 2]
```diff
@@ -390,12 +391,16 @@ impl<EK: KvEngine> Runner<EK> {
     /// Returns true if task is consumed. Failure is considered consumed.
     fn process_destroy_task(logger: &Logger, registry: &TabletRegistry<EK>, path: &Path) -> bool {
         match EK::locked(path.to_str().unwrap()) {
-            Err(e) => warn!(
-                logger,
-                "failed to check whether the tablet path is locked";
-                "err" => ?e,
-                "path" => path.display(),
-            ),
+            Err(e) if !e.to_string().contains("No such file or directory") => {
```
Comment (tonyxuqqi) [inline components/raftstore-v2/src/worker/tablet.rs:394]: Is there other better way to check if it's "no such file" error as I think error message may be changed over time?

Comment (tabokie) [reply components/raftstore-v2/src/worker/tablet.rs:394]: It's RocksDB error message, no other way.

**Review by tonyxuqqi — APPROVED** [review]

File: components/raftstore-v2/src/operation/command/admin/merge/prepare.rs [hunk 3]
```diff
@@ -345,7 +401,7 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
 
         let status = &mut self.merge_context_mut().prepare_status;
         // Shouldn't enter this call if trim check is already underway.
-        assert!(status.is_none());
+        assert!(status.is_none(), "{:?}", status);
```
Comment (overvenus) [inline components/raftstore-v2/src/operation/command/admin/merge/prepare.rs:404]: Is there any region id or peer id in `status`?

File: tests/integrations/config/mod.rs [hunk 4]
```diff
@@ -902,22 +902,6 @@ fn test_do_not_use_unified_readpool_with_legacy_config() {
     assert!(!cfg.readpool.is_unified_pool_enabled());
 }
 
-#[test]
-fn test_block_cache_backward_compatible() {
```
Comment (overvenus) [inline tests/integrations/config/mod.rs:906]: Why remove the test?

**Review by overvenus — COMMENTED** [review]
Rest LGTM

**Review by overvenus — APPROVED** [review]
